xmodule/library_content_block.py

```diff
@@ -668,15 +701,23 @@ def source_library_values(self):
 
     def editor_saved(self, user, old_metadata, old_content):  # lint-amnesty, pylint: disable=unused-argument
         """
-        If source_library_id or capa_type has been edited, refresh_children automatically.
+        If source_library_id is empty, clear source_library_version and children.
         """
-        old_source_library_id = old_metadata.get('source_library_id', [])
-        if (old_source_library_id != self.source_library_id or
-                old_metadata.get('capa_type', ANY_CAPA_TYPE_VALUE) != self.capa_type):
-            try:
+        if not self.source_library_id:
+            self.children = []  # lint-amnesty, pylint: disable=attribute-defined-outside-init
+            self.source_library_version = ""
+        else:
+            self.source_library_version = str(self.tools.get_library_version(self.source_library_id))
+
+    def post_editor_saved(self):
+        """
+        If xblock has been edited, refresh_children automatically.
```

Comment:
Open discussion: https://github.com/openedx/edx-platform/pull/33057#discussion_r1307945409

Reply:
So in summary: the post_editor_saved event is fired after the editor is saved, the editor_saved event fires when the author presses the save button. I am now sure of this, and can see the benefits of adding this lifecycle method, but am open to suggestions as to how we want to refactor it out.